### ISSUE TITLE
Cryptic error messages on Minizinc 2.5.2

### ISSUE THREAD

**Issue by etzhaim** [MiniZinc/libminizinc#437]
I have Minizinc 2.5.2 installed as a snap on Ubuntu 18.04 and 20.04. After upgrading from 2.4.x I get the following error messages on both OSes:

```
$ minizinc project/misc/oopt/findpin2.mzn 
MiniZinc: evaluation error: 
  /home/etzhaim/project/misc/oopt/findpin2.mzn:51:
  in call 'atwrongplace'
  /home/etzhaim/project/misc/oopt/findpin2.mzn:17:
  in let expression
  /home/etzhaim/project/misc/oopt/findpin2.mzn:18:
  in array comprehension expression
    with n = 1
  in call 'digit'
  /home/etzhaim/project/misc/oopt/findpin2.mzn:14:
  in binary 'mod' operator expression
  in binary 'div' operator expression
  /snap/minizinc/263/share/minizinc/std//stdlib/stdlib_math.mzn:106:
  in if-then-else expression
  in call 'mzn_in_root_context'
  internal error: missing builtin 'mzn_in_root_context'

$ minizinc --version
MiniZinc to FlatZinc converter, version 2.5.2, build 213472776
Copyright (C) 2014-2020 Monash University, NICTA, Data61
```

Here is a link to the MiniZinc model: https://pastebin.com/pNtsBzrk

Thanks in advance.

**Comment by Dekker1** [comment]
I believe this issue was resolved on the develop branch: 227ce0893e8d8e22967e0c039e5f704c14ffda3f
A test case to prevent this from happening in the future was added as well: ca334e5449a6e8177a3be46cbb4cb4408d3251bb

**Comment by cyderize** [comment]
This should now be fixed in [2.5.3](https://github.com/MiniZinc/MiniZincIDE/releases/tag/2.5.3).
